src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRunCodegenNodeFactory.cs

```diff
@@ -390,8 +457,35 @@ public IEnumerable<MethodWithGCInfo> EnumerateCompiledMethods()
             }
         }
 
-        private readonly Dictionary<ReadyToRunFixupKind, Dictionary<TypeAndMethod, MethodFixupSignature>> _methodSignatures =
-            new Dictionary<ReadyToRunFixupKind, Dictionary<TypeAndMethod, MethodFixupSignature>>();
+        private class MethodFixupKey : IEquatable<MethodFixupKey>
```

Comment:
@MichalStrehovsky would it be better to move these key structures into separate .cs files?

Reply:
I don't have a strong opinion. In the full AOT compiler we  dump them at the end of the .cs file to get them out of the way because they're not particularly interesting. It's a grab-bag of unrelated things, but that's the entire NodeFactory in general.

They're `private` nested types so that they don't leak into IntelliSense in places.

And I've just noticed... they should be `struct` not `class`, so that we don't have a GC allocation every time we look for one.

Reply:
I used to put them next to the point of use but I'm not against cleaning that up somehow by moving them to the end of the file or to a separate file.